src/weighted_graph.py

```diff
@@ -0,0 +1,191 @@
+        try:
+            del self._gdict[n1][n2]
+        except KeyError:
+            raise KeyError("That edge does not exist.")
+
+    def has_node(self, n):
+        """Return True if node n exists in the weighted graph."""
+        if n in self._gdict:
+            return True
+        return False
+
+    def neighbors(self, n):
+        """Return a list of all neighbors of node n."""
+        try:
+            return list(self._gdict[n])
+        except KeyError:
+            raise KeyError("Node {} does not exist".format(n))
+
+    def adjacent(self, n1, n2):
+        """Return true if there is an edge connecting n1 and n2."""
+        if n1 in self._gdict:
+            if n2 in self._gdict:
+                if n2 in self._gdict[n1]:
+                    return True
+                return False
+            else:
+                raise KeyError("The node {} is not in the graph.".format(n2))
+        else:
+            raise KeyError("The node {} is not in the graph.".format(n1))
+
+    def depth_first_traversal(self, start, path=None):
+        """Traverse a weighted graph depth first."""
+        if start not in self._gdict:
+            raise KeyError(start, "not in graph.")
+        if path is None:
+            path = []
+        path.append(start)
+        for neighbor in list(self._gdict[start]):
+            if neighbor not in path:
+                self.depth_first_traversal(neighbor, path)
+
+        return path
+
+    def breadth_first_traversal(self, start):
+        """Traverse a weighted graph breadth first."""
+        if start not in self._gdict:
+            raise KeyError(start, "not in graph.")
+        path = []
+        q = Queue()
+        q.enqueue(start)
+
+        while len(q) > 0:
+            current = q.dequeue()
+            if current not in path:
+                path.append(current)
+
+            for neighbor in list(self._gdict[current]):
+                if neighbor not in path:
+                    q.enqueue(neighbor)
+
+        return path
+
+    def shortest_dijkstra(self, start, target):
+        """Use Dijkstra's algorithm to find the shortest path from start to target."""
+        # import pdb; pdb.set_trace()
+        distance = {}
+        path_weights = {start: (None, 0)}
+        for key in self._gdict:
+            distance[key] = inf
+        distance[start] = 0
+
+        while distance:
+            curr = min(distance, key=distance.get)
+
+            for neighbor in self._gdict[curr]:
+                temp_dist = distance[curr] + self._gdict[curr][neighbor]
+
+                if neighbor in distance and temp_dist < distance[neighbor]:
+                    distance[neighbor] = temp_dist
+                    path_weights[neighbor] = (curr, temp_dist)
+
+            del distance[curr]
+
+        path = []
+        prev = target
+        while prev is not None:
+            path.append(prev)
+            prev = path_weights[prev][0]
+        return list(reversed(path))
```

Comment:
Really nice job here.  The only thing else I think you should do is raise a ValueError if there is no path from start to finish